text/0000-let-else.md

```diff
@@ -381,19 +404,76 @@ let AnEnum::Variant1(a) = x else {
 };
 ```
 
-Another potential alternative for fall-back which could be added with an additional keyword as a future extension:
+Another potential alternative for fall-back:
 
 ```rust
 enum AnEnum {
     Variant1(u32),
     Variant2(String),
 }
 
-let AnEnum::Variant1(a) = x else assign a {
-    a = 42;
+let Ok(a) = x else match {
+    Err(e) => return Err(e.into()),
+}
+```
+
+### Assign to outer scope from `match`
+
+Another alternative is to allow assigning to the outer scope from within a `match`.
+
+```rust
+match thing {
+  Happy(x) => let x, // Assigns x to outer scope.
+  Sad(y) => return Err(format!("We were sad because of {}", y)),
+  Tragic(z) => return Err(format!("We cried hard because of {}", z)),
+}
+```
+
+However this is not an obvious opposite io if-let, and would introduce an entirely new positional meaning of `let`.
+
+### `||` in pattern-matching
+
+A more complex, more flexible, but less obvious alternative is to allow `||` in any pattern matches as a fall-through match case fallback.
+Such a feature would likely interact more directly with [if-let-chains][], but could also be use to allow refutable patterns in let statements
+by covering every possible variant of an enum (possibly by use of a diverging fallback block similar to `_` in `match`).
+
+For example, covering the use-case of let-else:
+```rust
+let Some(x) = a || { return; };
+```
+
+With a fallback:
+```rust
+let Some(x) = a || b || { return; };
+```
+
+Combined with `&&` as proposed in if-let-chains, constructs such as the following are conceivable:
+
+```rust
+let Enum::Var1(x) = a || b || { return anyhow!("Bad x"); } && let Some(z) = x || y;
```

Comment:
This requires that `y: !`, which means that the entire line is dead code. Probably there should be another diverging expression at the end like `break`